[Feature](aggregate) support generic aggregate functions on pre-aggregate model

BiteTheDDDDt opened the pull request:
# Proposed changes

support generic aggregate functions on pre-aggregate model 

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

BiteTheDDDDt commented:
run buildall

hello-stephen commented:
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.15 seconds
 stream load tsv:          423 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230414140259_clickbench_pr_129219.html

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run feut

BiteTheDDDDt commented:
run p0

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt committed:
some fe prowork of generic aggregation

fix

fix

fix

support generic aggregate functions on pre-aggregate model

update

fix

fix

fix

update

fix

fix

update

fix

fix

fix

fix

update

fix

fix

update

fix

tmp

update

update

update

update

update

update

update

fix

fix

tmp

update

update

fix

update cases

update

update

reset

update

fix

update

rename

update

update

format

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt commented:
run buildall

BiteTheDDDDt committed:
update

BiteTheDDDDt committed:
update

BiteTheDDDDt committed:
update

BiteTheDDDDt committed:
update
